File: ios/VoiceHostSoftphone/Models/CallRecord.swift
import Foundation

struct CallRecord: Identifiable, Codable, Equatable {
    enum Direction: String, Codable {
        case incoming
        case outgoing
    }

    enum Result: String, Codable {
        case completed
        case missed
        case declined
        case failed
        case cancelled
    }

    let id: UUID
    let direction: Direction
    let number: String
    let displayName: String?
    let startedAt: Date
    let duration: TimeInterval
    let result: Result

    init(
        id: UUID = UUID(),
        direction: Direction,
        number: String,
        displayName: String? = nil,
        startedAt: Date,
        duration: TimeInterval,
        result: Result
    ) {
        self.id = id
        self.direction = direction
        self.number = number
        self.displayName = displayName
        self.startedAt = startedAt
        self.duration = duration
        self.result = result
    }

    var durationText: String? {
        guard duration > 0 else { return nil }
        let total = Int(duration.rounded(.down))
        return String(format: "%d:%02d", total / 60, total % 60)
    }

    var resultText: String {
        switch result {
        case .completed: return direction == .incoming ? "Incoming" : "Outgoing"
        case .missed: return "Missed"
        case .declined: return "Declined"
        case .failed: return "Failed"
        case .cancelled: return "Cancelled"
        }
    }
}

File: flutter/lib/models/call_record.dart
enum CallDirection { incoming, outgoing }

enum CallResult { completed, missed, declined, failed, cancelled }

class CallRecord {
  CallRecord({
    String? id,
    required this.direction,
    required this.number,
    this.displayName,
    required this.startedAt,
    required this.durationSeconds,
    required this.result,
  }) : id = id ?? '${startedAt.microsecondsSinceEpoch}-${number.hashCode}';

  final String id;
  final CallDirection direction;
  final String number;
  final String? displayName;
  final DateTime startedAt;
  final int durationSeconds;
  final CallResult result;

  String get durationText {
    final minutes = durationSeconds ~/ 60;
    final seconds = durationSeconds % 60;
    return '$minutes:${seconds.toString().padLeft(2, '0')}';
  }

  String get resultText => switch (result) {
        CallResult.completed =>
          direction == CallDirection.incoming ? 'Incoming' : 'Outgoing',
        CallResult.missed => 'Missed',
        CallResult.declined => 'Declined',
        CallResult.failed => 'Failed',
        CallResult.cancelled => 'Cancelled',
      };

  Map<String, dynamic> toJson() => {
        'id': id,
        'direction': direction.name,
        'number': number,
        'displayName': displayName,
        'startedAt': startedAt.toIso8601String(),
        'durationSeconds': durationSeconds,
        'result': result.name,
      };

  factory CallRecord.fromJson(Map<String, dynamic> json) => CallRecord(
        id: json['id'] as String?,
        direction: CallDirection.values.byName(json['direction'] as String),
        number: json['number'] as String,
        displayName: json['displayName'] as String?,
        startedAt: DateTime.parse(json['startedAt'] as String),
        durationSeconds: (json['durationSeconds'] as num?)?.toInt() ?? 0,
        result: CallResult.values.byName(json['result'] as String),
      );
}
